include/crow/multipart.h

```diff
@@ -6,44 +6,94 @@
 
 #include "crow/http_request.h"
 #include "crow/returnable.h"
+#include "crow/ci_map.h"
 
 namespace crow
 {
+
     /// Encapsulates anything related to processing and organizing `multipart/xyz` messages
     namespace multipart
     {
+
         const std::string dd = "--";
 
         /// The first part in a section, contains metadata about the part
         struct header
         {
-            std::pair<std::string, std::string> value;           ///< The first part of the header, usually `Content-Type` or `Content-Disposition`
+            std::string value;                                   ///< The first part of the header, usually `Content-Type` or `Content-Disposition`
             std::unordered_map<std::string, std::string> params; ///< The parameters of the header, come after the `value`
+
+            inline int i() const { return std::stoi(value); }    ///< Return \ref value as integer
+            inline double d() const { return std::stod(value); } ///< Return \ref value as double
+
+            operator int() const { return i(); }
+            operator double() const { return d(); }
         };
 
+        /// Multipart header map (key is header key).
+        using mph_map = std::unordered_multimap<std::string, header, ci_hash, ci_key_eq>;
+
+        /// Find and return the value object associated with the key. (returns an empty class if nothing is found)
+        template<typename O, typename T>
+        inline const O& get_header_value_object(const T& headers, const std::string& key)
+        {
+            if (headers.count(key))
+            {
+                return headers.find(key)->second;
+            }
+            static O empty;
+            return empty;
+        }
+
+        /// Same as \ref get_header_value_Object() but for \ref multipart.header
+        template<typename T>
+        inline const header& get_header_object(const T& headers, const std::string& key)
+        {
+            return get_header_value_object<header>(headers, key);
+        }
+
         ///One part of the multipart message
 
         ///
         /// It is usually separated from other sections by a `boundary`
         struct part
         {
-            //TODO(EDev): restructure this to an `unordered_map<string, header>` with string being `header::value.first`
-            std::vector<header> headers; ///< (optional) The first part before the data, Contains information regarding the type of data and encoding
-            std::string body;            ///< The actual data in the part
+            mph_map headers;  ///< (optional) The first part before the data, Contains information regarding the type of data and encoding
+            std::string body; ///< The actual data in the part
+
+            inline int i() const { return std::stoi(body); }    ///< Return \ref body as integer
```

Comment:
Why do you create those two methods? Wouldn't it be better to just call stoi/stod in operator?

Reply:
Fair enough